BenchmarkVCFs/CompareBenchmarks.wdl

```diff
@@ -0,0 +1,428 @@
+        self.file = file
+        self.separator = sep
+
+    def cells(self, cells:list):
+        self.file.write(self.separator.join([str(cell) for cell in cells]))
+        return self
+    def sep(self):
+        self.file.write(self.separator)
+        return self
+    def new_line(self):
+        self.file.write('\n')
+        return self
+
+def write_header(output:ChainableOutput, unique_sample_ids, unique_configurations, deltas, include_counts):
+    # Line 1
+    output.cells([''] * 4).sep()
+    if include_counts:
+        output.cells(['TP'] + [''] * (len(unique_sample_ids) * (len(unique_configurations) + len(deltas)) - 1)).sep()
+        output.cells(['FP'] + [''] * (len(unique_sample_ids) * (len(unique_configurations) + len(deltas)) - 1)).sep()
+        output.cells(['FN'] + [''] * (len(unique_sample_ids) * (len(unique_configurations) + len(deltas)) - 1)).sep()
+    output.cells(['Precision'] + [''] * (len(unique_sample_ids) * (len(unique_configurations) + len(deltas)) - 1)).sep()
+    output.cells(['Sensitivity'] + [''] * (len(unique_sample_ids) * (len(unique_configurations) + len(deltas)) - 1)).sep()
+    output.cells(['F-Measure'] + [''] * (len(unique_sample_ids) * (len(unique_configurations) + len(deltas)) - 1)).sep()
+    output.cells([''] * 2).new_line()
+
+    # Line 2
+    output.cells([''] * 4)
+    for _ in (['TP', 'FP', 'FN'] if include_counts else []) + ['Precision', 'Sensitivity', 'F-Measure']:
+        for sample_id in unique_sample_ids:
+            output.sep().cells([sample_id] + [''] * (len(unique_configurations) + len(deltas) - 1))
+    output.sep().cells([''] * 2).new_line()
+
+    # Line 3
+    output.cells([''] * 4)
+    for metric in (['TP', 'FP', 'FN'] if include_counts else []) + ['Precision', 'Sensitivity', 'F-Measure']:
+        for sample_id in unique_sample_ids:
+            # Write configurations themselves
+            for configuration in unique_configurations:
+                output.sep().cells([configuration])
+            # Write deltas
+            for delta_pair in deltas:
+                if metric in ['TP', 'FP', 'FN']:
+                    output.sep().cells([f'delta({unique_configurations[delta_pair[1]]}-{unique_configurations[delta_pair[0]]})'])
+                else:
+                    output.sep().cells([f'delta%({unique_configurations[delta_pair[1]]}-{unique_configurations[delta_pair[0]]})'])
+    output.sep().cells([''] * 2).new_line()
+
+def get_value_from_table(data, sample_id, configuration, stratifier, var_type, column):
+    return data.loc[(data['sample_id'] == sample_id) & (data['configuration'] == configuration) & (data['Stratifier'] == stratifier) & (data['Type'] == var_type)].iloc[0][column]
+
+def write_stratifier(output:ChainableOutput, stratifier:str, data:pd.DataFrame, unique_sample_ids:list, unique_configurations:list, deltas:list, include_counts):
+    for var_type in ['SNP', 'INDEL', 'all']:
+        # Placeholder to print percentage of genome for each stratification
```

Comment:
does this placeholder get filled by the workflow (I'm guessing no?) or do you fill it in later by hand?

Reply:
Initially I've been doing that manually, but I started working on a task that calculates the genome territory for each stratifier, which can then be used here. That's a work in progress at this point though